cmake/add_enclave.cmake

```diff
@@ -101,3 +147,91 @@ function(add_enclave)
   # Add a custom target with `ALL` semantics so these targets are always built.
   add_custom_target(${ENCLAVE_TARGET}_signed_target ALL DEPENDS ${SIGNED_LOCATION})
 endfunction()
+
+macro(add_enclave_optee)
+   set(options CXX)
+   set(oneValueArgs TARGET UUID KEY)
+   set(multiValueArgs SOURCES)
+   cmake_parse_arguments(ENCLAVE
+     "${options}"
+     "${oneValueArgs}"
+     "${multiValueArgs}"
+     ${ARGN})
+
+  # Set up the linker flags exactly as we need them such that the resulting
+  # binary be compatible with OP-TEE's loader.
+  set(CMAKE_SHARED_LIBRARY_LINK_C_FLAGS)
+  set(CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS)
+  set(CMAKE_EXE_EXPORTS_C_FLAG)
+
+  string(REPLACE "gcc" "ld" LINKER ${CMAKE_C_COMPILER})
```

Comment:
Seems easy to break...

Reply:
If we ever wish to build TA's with a toolchain other than GCC, then yes. I have yet to see anyone try to build TA's with clang, so I would argue that this solution is sufficient in the short-term, seeing as GCC binaries follow the pattern `<toolchain-prefix>-<tool-name>`.